src/PackagePublisher/Program.cs

```diff
@@ -0,0 +1,246 @@
+
+            try
+            {
+                return app.Execute(args);
+            }
+            catch (Exception ex)
+            {
+                Log.WriteError(ex.ToString());
+                return 1;
+            }
+        }
+
+        private static bool AreValidParameters()
+        {
+            if (!_feedToUploadTo.HasValue() || string.IsNullOrEmpty(_feedToUploadTo.Value()))
+            {
+                Log.WriteError("Feed is required.");
+                return false;
+            }
+
+            _apiKey = Environment.GetEnvironmentVariable(ApiKeyEnvironmentVariableName);
+            if (string.IsNullOrEmpty(_apiKey))
+            {
+                Log.WriteError("Api key is required to publish to the feed. Set the environment " +
+                    $"variable {ApiKeyEnvironmentVariableName} before inovking this tool again.");
+                return false;
+            }
+
+            if (!_packagesDirectory.HasValue() && !_packagePath.HasValue())
+            {
+                Log.WriteError("Provide a directory path with the packages or a full file path to package(s)");
+                return false;
+            }
+            else if (_packagesDirectory.HasValue() && _packagePath.HasValue())
+            {
+                Log.WriteError("Both package directory and package path cannot be supplied. Specify either of them only.");
+                return false;
+            }
+            else
+            {
+                if (_packagesDirectory.HasValue() && string.IsNullOrEmpty(_packagesDirectory.Value()))
+                {
+                    Log.WriteError("Invalid value for package directory.");
+                    return false;
+                }
+
+                if (_packagePath.HasValue() && _packagePath.Values.Count == 0)
+                {
+                    Log.WriteError("Invalid value for package path.");
+                    return false;
+                }
+            }
+
+            if (_maxRetryCountOption.HasValue())
+            {
+                _maxRetryCount = int.Parse(_maxRetryCountOption.Value());
+            }
+
+            return true;
+        }
+
+        private static IEnumerable<PackageInfo> ProcessPackages(this IEnumerable<FileInfo> packageFiles)
+        {
+            return packageFiles
+                .Select(fileInfo =>
+                {
+                    using (var fileStream = File.OpenRead(fileInfo.FullName))
+                    using (var reader = new PackageArchiveReader(fileStream))
+                    {
+                        return new PackageInfo()
+                        {
+                            Identity = reader.GetIdentity(),
+                            PackagePath = fileInfo.FullName
+                        };
+                    }
+                });
+        }
+
+        private static async Task PublishToFeedAsync()
+        {
+            using (var semaphore = new SemaphoreSlim(initialCount: 4, maxCount: 4))
```

Comment:
We've historically seen timeouts increase when we bumped up this number. Although that was when we were publishing to `www.myget.org` as opposed to `dotnet.myget.org`.